src/Compilers/CSharp/Test/Symbol/Symbols/LocalFunctionTests.cs

```diff
@@ -189,5 +189,181 @@ public void M()
+        }
+
+        [Fact]
+        public void StaticLocalFunction_CapturingMethodGroup2()
+        {
+            CreateCompilation("""
+                using System;
+
+                var c = new C();
+                LocalFunc();
+                NonStatic();
+
+                static void LocalFunc()
+                {
+                    var x = Extensions.MyExtension;
+                    var y = new Func<C, string>(Extensions.MyExtension);
+                }
+
+                void NonStatic()
+                {
+                    Action f = static () =>
+                    {
+                        var x = Extensions.MyExtension;
+                        var y = new Func<C, string>(Extensions.MyExtension);
+                    };
+                }
+
+                public class C
+                {
+                }
+
+                public static class Extensions
+                {
+                    public static string MyExtension(this C c)
+                        => string.Empty;
+                }
+                """).VerifyDiagnostics().VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void StaticLocalFunction_CapturingMethodGroup3()
+        {
+            CreateCompilation("""
+                using System;
+                public class Base { }
+
+                public class C : Base
+                {
+                    public void M()
+                    {
+                        LocalFunc();
+                        NonStatic();
+
+                        static void LocalFunc()
+                        {
+                            var x1 = this.MyExtension;
+                            var x2 = new Func<string>(this.MyExtension);
+                            var y1 = base.MyExtension;
+                            var y2 = new Func<string>(base.MyExtension);
+                        }
+                
+                        void NonStatic()
+                        {
+                            Action f = static () =>
+                            {
+                                var x1 = this.MyExtension;
+                                var x2 = new Func<string>(this.MyExtension);
+                                var y1 = base.MyExtension;
+                                var y2 = new Func<string>(base.MyExtension);
+                            };
+                        }
+                    }
+                }
+
+                internal static class Extensions
+                {
+                    public static string MyExtension(this Base c)
+                        => string.Empty;
+                }
+                """).VerifyDiagnostics(
+                    // (13,22): error CS8422: A static local function cannot contain a reference to 'this' or 'base'.
+                    //             var x1 = this.MyExtension;
+                    Diagnostic(ErrorCode.ERR_StaticLocalFunctionCannotCaptureThis, "this").WithLocation(13, 22),
```

Comment:
Are we missing more `ERR_StaticLocalFunctionCannotCaptureThis` diagnostics? I'm not sure.

Reply:
> Are we missing more ERR_StaticLocalFunctionCannotCaptureThis diagnostics? I'm not sure.

Is there a specific concern here?

Reply:
No, I asked just to make sure that nothing is wrong